inspire_dojson/hep/model.py

```diff
@@ -83,12 +86,46 @@ def convert_curated(record, blob):
     return record
 
 
+def ensure_ordered_figures(record, blob):
+    ordered_figures_dict = {}
+    unordered_figures_list = []
+
+    for figure in record.get('figures', []):
+        order = figure.pop('order')
+        if order:
+            ordered_figures_dict[order] = figure
+        else:
+            unordered_figures_list.append(figure)
+
+    record['figures'] = [value for key, value in sorted(six.iteritems(ordered_figures_dict))]
+    record['figures'].extend(unordered_figures_list)
+    return record
+
+
+def ensure_unique_documents_and_figures(record, blob):
+    def dupliactes(elements):
+        duplicate_keys_list = []
+        for index, element in enumerate(elements):
+            if element:
```

Comment:
why is this `if` needed?

Reply:
Because we clean the record after we have some times list [None] and then it crashes. I wanted to keep the cleaning step for last otherwise I can move the check after cleaning. What do you think?

Reply:
OK then.